p11-kit/list-objects.c

```diff
@@ -274,56 +272,59 @@ print_object (p11_list_printer *printer,
 
 	p11_kit_iter_get_attributes (iter, attrs, n_attrs);
 	for (i = 0; i < n_attrs; ++i) {
+		p11_buffer buffer;
+
 		if (attrs[i].pValue != NULL_PTR ||
 		    attrs[i].ulValueLen == CK_UNAVAILABLE_INFORMATION)
 			continue;
 
-		if (attrs[i].ulValueLen >= SIZE_MAX) {
-			p11_message (_("allocation would result in overflow"));
+		if (!p11_buffer_init_null (&buffer, attrs[i].ulValueLen) ||
+		    !p11_buffer_append (&buffer, attrs[i].ulValueLen)) {
```

Comment:
I get it now. The `p11_buffer_append` function doesn't actually append data, it just moves the length pointer... :smiling_face_with_tear: